### PR TITLE
RxJS rewrite

### PR DESCRIPTION
**Pull request by alexanderGugel**
This is a WIP. It's not done yet. I have been playing around with this in a separate project for a while now and it's time to put it into the actual project.

Currently it's not working (updated RxJS). There is some missing functionality, but overall I think RxJS simplifies the overall flow.

I'm trying to get this in as soon as possible.

### CODE REVIEW

**Comment by STRML** [comment]
@alexanderGugel If we're looking to contribute to ied, should we build off of this branch rather than the existing code?

**Comment by alexanderGugel** [comment]
Yup, definitely this branch. RxJS is the future!

File: src/run_cmd.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    const scripts = [
+      `pre${scriptName}`,
+      scriptName,
+      `post${scriptName}`
+    ]
+
+    const scriptCmds = scripts.map((scriptName) => {
+      return pkg.scripts[scriptName]
+    }).filter((scriptCmd) => {
+      return scriptCmd
```
Comment (STRML) [inline src/run_cmd.js]: This should throw instead, no? npm does if you try to run a script that doesn't exist.

**Comment by STRML** [comment]
@alexanderGugel I know you're probably sidetracked with the Node6 debacle - anything I can do to help so we can get this out & move forward?

**Comment by alexanderGugel** [comment]
`next` landed.

**Pull request closed without merging**